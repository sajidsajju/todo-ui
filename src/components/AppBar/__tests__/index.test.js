import React from "react";
import { render, screen } from "@testing-library/react";

import TodoAppBar from "../";

describe("<AppBar  />", () => {
  const userName = "Sajid Ahmed";
  it("should match the snapshot", () => {
    const rendered = render(<TodoAppBar userName={userName} />);

    expect(rendered.container.firstChild).toMatchSnapshot();
  });

  it("should match the title of the AppBar", () => {
    render(<TodoAppBar userName={userName} />);

    expect(screen.getByText(`Welcome,`)).toBeInTheDocument();
  });

  it("should match the signout button", () => {
    render(<TodoAppBar userName={userName} />);

    expect(screen.getByText("Logout")).toBeInTheDocument();
  });
});
