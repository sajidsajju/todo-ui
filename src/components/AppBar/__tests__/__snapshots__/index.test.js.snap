// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppBar  /> should match the snapshot 1`] = `
<div
  class="makeStyles-grow-1"
>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-2 mui-fixed MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        class="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
      >
        Welcome, 
      </h6>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-logout-4"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Logout
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>
</div>
`;
